imageio/core/imopen.py

```diff
@@ -0,0 +1,477 @@
+from numbers import Number
+import re
+import numpy as np
+
+from .format import FormatManager, MODENAMES
+from .request import Request, RETURN_BYTES
+
+
+class imopen(object):
+    """ Open a URI and return a plugin instance that can read its content.
+
+    Imopen is the plugin manager of imopen. It acts as a factory for Plugin
+    objects implementing the v3.0 API. It is the core of the v3.0 API and the
+    functional API provides convenient wrappers around this object.
```

Comment:
This could need some work to make it clear for users that simply want to know when and how to use this function :)

Reply:
I updated the paragraph in question. Let me know if you had something else in mind.